R/ml_feature_robust_scaler.R

```diff
@@ -0,0 +1,126 @@
+                             lower = 0.25, upper = 0.75, with_centering = TRUE,
+                             with_scaling = TRUE, relative_error = 0.001,
+                             uid = random_string("ft_robust_scaler_"), ...) {
+  check_dots_used()
+  UseMethod("ft_robust_scaler")
+}
+
+ml_robust_scaler <- ft_robust_scaler
+
+#' @export
+ft_robust_scaler.spark_connection <- function(x, input_col = NULL, output_col = NULL,
+                                              lower = 0.25, upper = 0.75, with_centering = TRUE,
+                                              with_scaling = TRUE, relative_error = 0.001,
+                                              uid = random_string("ft_robust_scaler_"), ...) {
+  .args <- list(
+    input_col = input_col,
+    output_col = output_col,
+    lower = lower,
+    upper = upper,
+    with_centering = with_centering,
+    with_scaling = with_scaling,
+    relative_error = relative_error,
+    uid = uid
+  )%>%
+    c(rlang::dots_list(...)) %>%
+    validator_ml_robust_scaler()
+
+  estimator <- spark_pipeline_stage(
+    x, "org.apache.spark.ml.feature.RobustScaler",
+    input_col = .args[["input_col"]], output_col = .args[["output_col"]], uid = .args[["uid"]]
+  ) %>%
+    invoke("setLower", .args[["lower"]]) %>%
+    invoke("setUpper", .args[["upper"]]) %>%
+    invoke("setWithCentering", .args[["with_centering"]]) %>%
+    invoke("setWithScaling", .args[["with_scaling"]]) %>%
+    invoke("setRelativeError", .args[["relative_error"]]) %>%
+    new_ml_robust_scaler()
+
+  estimator
+}
+
+#' @export
+ft_robust_scaler.ml_pipeline <- function(x, input_col = NULL, output_col = NULL,
+                                         lower = 0.25, upper = 0.75, with_centering = TRUE,
+                                         with_scaling = TRUE, relative_error = 0.001,
+                                         uid = random_string("ft_robust_scaler_"), ...) {
+
+  stage <- ft_robust_scaler.spark_connection(
+    x = spark_connection(x),
+    input_col = input_col,
+    output_col = output_col,
+    lower = lower,
+    upper = upper,
+    with_centering = with_centering,
+    with_scaling = with_scaling,
+    relative_error = relative_error,
+    uid = uid,
+    ...
+  )
+
+  ml_add_stage(x, stage)
+}
+
+#' @export
+ft_robust_scaler.tbl_spark <- function(x, input_col = NULL, output_col = NULL,
+                                        lower = 0.25, upper = 0.75, with_centering = TRUE,
+                                        with_scaling = TRUE, relative_error = 0.001,
+                                        uid = random_string("ft_robust_scaler_"), ...) {
+  stage <- ft_robust_scaler.spark_connection(
+    x = spark_connection(x),
+    input_col = input_col,
+    output_col = output_col,
+    lower = lower,
+    upper = upper,
+    with_centering = with_centering,
+    with_scaling = with_scaling,
+    relative_error = relative_error,
+    uid = uid,
+    ...
+  )
+
+  if (is_ml_transformer(stage))
```

Comment:
It doesn't seem like this can ever happen, but since this pattern is repeated all over the code base, I'll keep it for consistency.